crates/iceberg/src/avro/schema.rs

```diff
@@ -96,7 +98,13 @@ impl SchemaVisitor for SchemaToAvroSchema {
         _struct: &StructType,
         results: Vec<AvroSchemaOrField>,
     ) -> Result<AvroSchemaOrField> {
-        let avro_fields = results.into_iter().map(|r| r.unwrap_right()).collect();
+        let avro_fields: Vec<AvroRecordField> =
+            results.into_iter().map(|r| r.unwrap_right()).collect();
+
+        let mut lookup = BTreeMap::new();
```

Comment:
How about using `BTreeMap::from_iter()` instead of for-loop insert? `from_iter` uses bulk build internally which has better performance.

Reply:
Updated